torch/nn/modules/module.py

```diff
@@ -461,3 +461,10 @@ def __repr__(self):
             tmpstr = tmpstr + '  (' + key + '): ' + modstr + '\n'
         tmpstr = tmpstr + ')'
         return tmpstr
+
+    def __dir__(self):
+        _moduel_attrs = dir(self.__class__)
+        _parameters = self._parameters.keys()
+        _modules = self._modules.keys()
+        _attrs = self.__dict__.keys()
+        return _moduel_attrs + _parameters + _modules + _attrs
```

Comment:
I don't think we should replace `dir(self.__class__)` with `super(Module, self).__dir__()`, it will raise `AttributeError: 'super' object has no attribute '__dir__'` when calling `dir(Linear(10,2))`. 

When calling `dir(self.__class__)` I wanna got all methods of the current module. `Linear` for example, I wanna get method `reset_parameters` that is only available for `Linear`.

So I think maybe it should be implemented like this
```python
    def __dir__(self):
        _module_attrs = dir(self.__class__)
        _attrs = self.__dict__.keys()
        _parameters = self._parameters.keys()
        _modules = self._modules.keys()
        _buffers = self._buffers.keys()
        return _module_attrs + _attrs + _parameters + _modules + _buffers
```